lib/schemaUtils.js

```diff
@@ -1360,6 +1361,39 @@ module.exports = {
     return example;
   },
 
+  /**
+   * Identifies if the example.value should be used as value of the body
+   * This is needed because there are cases that the example object has a property
+   * called value like:  example = { id: 1, value: "someValue" }
+   * In this case the body value should be the whole example object and not the property called value
+   * The property value belongs to the example object not the example OAS spec property
+   * @param {*} schema bodyObject (media type) schema to use
+   * @param {*} example - Exampel took from the bodyObject
+   * @param {object} components - components defined in the OAS spec.
+   * @returns {boolean} Wheter to use the property value of the example as the value of the
+   * body data
+   */
+  useExampleValuePropAsValue(schema, example, components) {
+    let schemaProperties = [],
+      schemaObject,
+      schemaDataPath = '',
+      exampleProperties = [];
+    if (!schema) {
+      return true;
+    }
+    if (schema.$ref) {
+      schemaDataPath = formatDataPath(formatSchemaPathFromAJVErrorToConvertToDataPath(schema.$ref));
```

Comment:
@LuisTejedaS Why are we using this when there is an existing implementation that does the same things (also handle nested $refs). i.e. `getRefObject()`. Take a look at line no. 1433 for this branch to get more idea.